src/handlers.rs

```diff
@@ -28,6 +29,13 @@ pub struct GenericStringResponse {
     pub result: String,
 }
 
+pub const SESSION_VERSION_UPDATE_INTERVAL: i64 = 30 * 60;
+
+lazy_static::lazy_static! {
+
+    pub static ref SESSION_VERSIONS: Mutex<HashMap<String, (i64, String)>> = Mutex::new(HashMap::new());
```

Comment:
Should probably use `parking_lot::RwLock` to allow multiple concurrent readers since data is not going to be accessed for writing very often.